cmd/soci/commands/push.go

```diff
@@ -145,7 +148,41 @@ if they are available in the snapshotter's local content store.
 			} else {
 				dst.Client = authClient
 			}
+			existingIndexOption := cliContext.String(internal.ExistingIndexFlagName)
+			if ok := internal.ValidExistingIndexOptions[existingIndexOption]; !ok {
+				return fmt.Errorf("unexpected value for flag --existing-index: %s, expected types [%s, %s, %s, %s]",
+					existingIndexOption, internal.Warn, internal.Skip, internal.Allow, internal.Abort)
+			}
+			if existingIndexOption != internal.Allow {
+				if !quiet {
+					fmt.Println("checking if a soci index already exists in remote repository...")
+				}
+				client := fs.NewOCIArtifactClient(dst)
+				referrersDes, err := client.AllReferrers(ctx, ocispec.Descriptor{Digest: imgManifestDesc.Digest})
+				if err != nil && !errors.Is(err, fs.ErrNoReferrers) {
+					return fmt.Errorf("failed to fetch list of referrers: %w", err)
+				}
+				if len(referrersDes) > 0 {
+					var foundMessage string
+					if len(referrersDes) > 1 {
+						foundMessage = "multiple soci indices found in remote repository"
+					} else {
+						foundMessage = fmt.Sprintf("soci index found in remote repository with digest: %s", referrersDes[0].Digest.String())
+					}
+					switch existingIndexOption {
+					case internal.Abort:
+						return fmt.Errorf("[ERROR] %s: aborting pushing artifacts for all image manifests", foundMessage)
+					case internal.Warn:
+						fmt.Printf("[WARN] %s: pushing index anyway\n", foundMessage)
```

Comment:
`Warn` should also `continue` right?

Reply:
No we should still attempt to the push the index for the current image manifest.

Reply:
I see, just notice there is no return in case Warn...

Reply:
I think there are 2 things that we can do to make this easier to scan:
1) Swap the order of `Warn` and `Skip` so that the last case is the one that falls through. It makes the code a little more linear
2) Add a comment after this printf that explicitly calls out the fall through, e.g. `// Fall through and attempt to push the index anyway`